### PR TITLE
[SPARK-34308][SQL] Escape meta-characters in printSchema

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

Similar to SPARK-33690, this PR improves the output layout of `printSchema` for the case column names contain meta characters.
Here is an example.

Before:
```
scala> val df1 = spark.sql("SELECT 'aaa\nbbb\tccc\rddd\feee\bfff\u000Bggg\u0007hhh'")
scala> df1.printSchema
root
 |-- aaa
ddd	ccc
   eefff
        ggghhh: string (nullable = false)
```

After:
```
scala> df1.printSchema
root
 |-- aaa\nbbb\tccc\rddd\feee\bfff\vggg\ahhh: string (nullable = false)
```

### Why are the changes needed?

To avoid breaking the layout of `Dataset#printSchema`

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala [hunk 1]
```diff
@@ -271,4 +271,18 @@ private[spark] object SchemaUtils {
     }
     field._1
   }
+
+  /**
+   * @param str The string to be escaped.
+   * @return The escaped string.
+   */
+  def escapeMetaCharacters(str: String): String = {
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala:279]: How about moving this into `org.apache.spark.sql.catalyst.util.StringUtils` instead?

Comment (sarutak) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala:279]: Hmm, I intended to use this for escaping strings in schema so I implemented this in `SchemaUtils`.
In fact, it's used for such purpose for now.
If we find it's useful for other purposes, how about moving this into a different place in the future?

**Review by srowen — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala (hunk 1 above)
Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala:279]: sgtm

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**